### PR TITLE
Add the possibility to setup public ELB to be accessed via HTTPS

### PR DESCRIPTION
**Pull request by TBonnin**
This patch adds a CFN template parameter for the SSL certificate ARN
If one is specified, the public ELB will be configured to accept
requests via https on port 443

@satterly

### CODE REVIEW

File: src/lib/auth.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ exports.setup = function(express, app, config) {
       done(null, obj)
     })
 
-    var callbackUrl = 'http://' + config.host + '/auth/google/callback'
+    var callbackUrl = config.host + '/auth/google/callback'
```
Comment (TBonnin) [inline src/lib/auth.js]: The url scheme is now part of the host url (see https://github.com/guardian/elk-stack/pull/61/files#diff-fea7794cc929f6e1c59c3329bf062533R539)

File: cloudformation/ELK_Stack_Multi_AZ_in_Private_VPC.json [hunk 2]
```diff
@@ -528,9 +535,12 @@
...
                                 " -e s,@@LOGCABIN_HOST,",
-                                    { "Fn::If" : [ "HasDNS",
-                                        { "Fn::Join": [".", [ "kibana", {"Ref": "HostedZoneName"} ]] },
-                                        { "Fn::GetAtt": [ "ElkPublicLoadBalancer", "DNSName" ] }
+                                    { "Fn::Join": ["", 
+                                        [{ "Fn::If" : [ "HasSSLCertificate", "https://", "http://" ] },
+                                        { "Fn::If" : [ "HasDNS",
+                                            { "Fn::Join": [".", [ "kibana", {"Ref": "HostedZoneName"} ]] },
+                                            { "Fn::GetAtt": [ "ElkPublicLoadBalancer", "DNSName" ]}
+                                        ]}]
```
Comment (TBonnin) [inline cloudformation/ELK_Stack_Multi_AZ_in_Private_VPC.json]: I am not a huge fan of the nested `Fn:...` but I am not aware of any other way to do so with cfn template

File: src/lib/auth.js (hunk 1 above)
Comment (satterly) [inline src/lib/auth.js]: Are you sure this works?

File: cloudformation/ELK_Stack_Multi_AZ_in_Private_VPC.json (hunk 2 above)
Comment (satterly) [inline cloudformation/ELK_Stack_Multi_AZ_in_Private_VPC.json]: Me neither, for both.

**Comment by satterly** [comment]
👍

File: cloudformation/ELK_Stack_Multi_AZ_in_Private_VPC.json (hunk 2 above)
Comment (TBonnin) [inline cloudformation/ELK_Stack_Multi_AZ_in_Private_VPC.json]: 😞

**Pull request merged**